### PR TITLE
new resource `azurerm_spring_cloud_builder`

### PR DESCRIPTION
**Pull request by ms-henglu**
image
image

### CODE REVIEW

File: website/docs/r/spring_cloud_builder.html.markdown [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+}
+
+resource "azurerm_spring_cloud_builder" "example" {
+  name                    = "example"
+  spring_cloud_service_id = azurerm_spring_cloud_service.example.id
+
+  build_pack_groups {
+    name           = "mix"
+    build_pack_ids = ["tanzu-buildpacks/java-azure"]
+  }
```
Comment (katbyte) [inline website/docs/r/spring_cloud_builder.html.markdown:39]: should this be build_pack_group? so

```suggestion
  build_pack_group {
    name           = "mix"
    build_pack_ids = ["tanzu-buildpacks/java-azure"]
  }
  
   build_pack_group {
    name           = "mix"
    build_pack_ids = ["tanzu-buildpacks/java-azure"]
  }
```

?

**Comment by ms-henglu** [comment]
Thanks @katbyte , I've updated it, please take another look.

File: internal/services/springcloud/spring_cloud_builder_resource.go [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+						"id": {
+							Type:     pluginsdk.TypeString,
+							Required: true,
+						},
+
+						"version": {
+							Type:     pluginsdk.TypeString,
+							Required: true,
+						},
+					},
```
Comment (katbyte) [inline internal/services/springcloud/spring_cloud_builder_resource.go:91]: can we please properly validate all this?

File: website/docs/r/spring_cloud_builder.html.markdown [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+* `build_pack_ids` - (Optional) Specifies a list of the build pack's ID.
+
+---
+
+A `stack` block supports the following:
+
+* `id` - (Required) ID of the ClusterStack.
+
+* `version` - (Required) Version of the ClusterStack
```
Comment (katbyte) [inline website/docs/r/spring_cloud_builder.html.markdown:74]: and fix the grammar/punctuation here?

**Review by katbyte — CHANGES_REQUESTED** [review]

**Review by katbyte — APPROVED** [review]
LGTM 🌿

**Pull request merged**